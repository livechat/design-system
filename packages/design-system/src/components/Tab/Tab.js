import React from 'react';
import PropTypes from 'prop-types';
import classNames from 'classnames/bind';
import styles from './style.css';

const noop = () => {};
const cx = classNames.bind(styles);

const Tab = ({ onSelect, href, isSelected, children, description }) => {
  const Component = props => (href ? <a {...props} /> : <button {...props} />);

  return (
    <Component
      href={href}
      onClick={onSelect}
      className={cx({
        tab: true,
        'tab--selected': isSelected
      })}
    >
      {children}
      {description && (
        <span className={styles.tab__description}>({description})</span>
      )}
    </Component>
  );
};

Tab.propTypes = {
  children: PropTypes.node.isRequired,
  onSelect: PropTypes.func,
  description: PropTypes.string,
  href: PropTypes.string,
  isSelected: PropTypes.bool
};

Tab.defaultProps = {
  onSelect: noop,
  description: null,
  href: null,
  isSelected: false
};

export default Tab;
