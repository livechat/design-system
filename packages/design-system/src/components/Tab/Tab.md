Tabs make it easy to navigate between views within the same context. They should group similar content with different filters applied.

Give tab a clear label that describes its content.
Tab label should be set in sentence case, and should not exceed three words. Donâ€™t use icons in the tab labels.
```js
initialState = { selectedTab: 'agents' };
const items = [
  {id: 'agents', title: 'Agents', count: 1},
  {id: 'groups', title: 'Groups', count: 3}
];

<div>
  <TabsWrapper>
    <TabsList>
      {items.map(({id, count, title}) => (
        <Tab
          onSelect={() => setState({selectedTab: id})}
          key={id}
          isSelected={state.selectedTab === id}
          description={count}
        >
          {title}
        </Tab>
      ))}
    </TabsList>
  </TabsWrapper>
</div>            
```

```html
<div class="lc-tabs">
  <div class="lc-tabs__list">
    <button class="lc-tab lc-tab--selected">Agents<span class="lc-tab__description">(1)</span></button>
    <button class="lc-tab">Groups<span class="lc-tab__description">(3)</span></button>
  </div>
</div>
```