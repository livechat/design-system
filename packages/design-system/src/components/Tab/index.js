export { default as TabsWrapper } from './TabsWrapper';
export { default as TabsList } from './TabsList';
export { default as Tab } from './Tab';
