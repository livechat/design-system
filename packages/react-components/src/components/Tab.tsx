import * as React from 'react';
import cx from 'classnames';
import { Text } from './Text';

type HTMLProps =
  | (React.AnchorHTMLAttributes<HTMLAnchorElement> & { href: string })
  | (React.ButtonHTMLAttributes<HTMLButtonElement> & { href?: never });

export type ITabProps = HTMLProps & {
  description?: React.ReactNode;
  isSelected?: boolean;
};

const baseClass = 'lc-tab';

export const Tab: React.FC<ITabProps> = ({
  children,
  className,
  description,
  isSelected,
  ...restProps
}) => {
  return (
    <Text
      {...restProps}
      as={restProps.href ? 'a' : 'button'}
      size="md"
      bold={isSelected}
      className={cx(
        baseClass,
        className,
        isSelected && `${baseClass}--selected`
      )}
    >
      {children}
      {description && (
        <Text as="span" size="md" className={`${baseClass}__description`}>
          ({description})
        </Text>
      )}
    </Text>
  );
};
