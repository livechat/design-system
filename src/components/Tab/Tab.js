import * as React from 'react';
import * as PropTypes from 'prop-types';
import classNames from 'classnames/bind';
import styles from './style.scss';
import getMergedClassNames from '../../utils/getMergedClassNames';

const cx = classNames.bind(styles);

const Tab = ({
  children,
  className,
  description,
  href,
  isSelected,
  ...restProps
}) => {
  const Component = props => (href ? <a {...props} /> : <button {...props} />);

  const mergedClassNames = getMergedClassNames(
    cx({
      tab: true,
      'tab--selected': isSelected
    }),
    className
  );

  return (
    <Component {...restProps} href={href} className={mergedClassNames}>
      {children}
      {description && (
        <span className={styles.tab__description}>({description})</span>
      )}
    </Component>
  );
};

Tab.propTypes = {
  children: PropTypes.node.isRequired,
  className: PropTypes.string,
  description: PropTypes.oneOfType([
    PropTypes.string,
    PropTypes.number,
    PropTypes.node
  ]),
  href: PropTypes.string,
  isSelected: PropTypes.bool
};

Tab.defaultProps = {
  description: null,
  href: null,
  isSelected: false
};

export default Tab;
