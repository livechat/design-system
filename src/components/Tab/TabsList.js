import * as React from 'react';
import * as PropTypes from 'prop-types';
import styles from './style.scss';

const TabsList = ({ children }) => (
  <div className={styles.tabs__list}>{children}</div>
);

TabsList.propTypes = {
  children: PropTypes.node.isRequired
};

export default TabsList;
