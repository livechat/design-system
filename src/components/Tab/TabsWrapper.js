import * as React from 'react';
import * as PropTypes from 'prop-types';
import styles from './style.scss';

const TabsWrapper = ({ children }) => (
  <div className={styles.tabs}>{children}</div>
);

TabsWrapper.propTypes = {
  children: PropTypes.node.isRequired
};

export default TabsWrapper;
