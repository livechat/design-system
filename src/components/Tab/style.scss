@import '../../scss/variables/colors';

.tab {
  position: relative;
  display: flex;
  flex-direction: row;
  align-items: center;
  padding: 0 24px;
  border: none;
  background: none;
  outline: none;
  color: $font-primary-color;
  cursor: pointer;
  transition: color 0.25s;
  margin: 0 8px 11px 8px;
  font-size: 14px;
  overflow: visible;

  &__description {
    display: inline-block;
    color: #8e949a;
    opacity: 0.6;
    padding-left: 3px;
    transition: color 0.25s;
  }

  &::before {
    position: absolute;
    bottom: 0;
    left: 0;
    right: 0;
    z-index: 1;
    display: block;
    content: '';
    background-color: $primary-color;
    height: 0;
  }

  &--selected {
    font-weight: 700;

    .tab__description {
      color: #8e949a;
    }

    &::before {
      height: 4px;
      bottom: -12px;
    }
  }

  &:hover {
    color: #000;
    text-decoration: none;

    .tab__description {
      color: #000;
    }
  }
}

.tabs {
  position: relative;
  border-bottom: 1px solid $divider-color;

  &__list {
    position: relative;
    display: flex;
    flex-direction: row;
    font-size: 14px;
    margin: 0 -8px;
  }
}
