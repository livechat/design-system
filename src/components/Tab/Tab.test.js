import * as React from 'react';
import { mount } from 'enzyme';
import Tab from './Tab';

describe('Tab', () => {
  it('should render selected tab', () => {
    const component = mount(
      <Tab isSelected description="1">
        Agents
      </Tab>
    );

    expect(component).toMatchSnapshot();
  });
});
