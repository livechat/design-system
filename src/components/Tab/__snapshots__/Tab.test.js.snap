// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tab should render selected tab 1`] = `
<Tab
  description="1"
  href={null}
  isSelected={true}
>
  <Component
    className="tab tab--selected"
    href={null}
  >
    <button
      className="tab tab--selected"
      href={null}
    >
      Agents
      <span
        className="tab__description"
      >
        (
        1
        )
      </span>
    </button>
  </Component>
</Tab>
`;
